### DISCUSSION TITLE
Auto generate signed URLs via Postgres HTTP

### DISCUSSION THREAD

**Discussion by cspecter** [supabase/supabase#1544] [Questions] [answered]
I am wondering if would be possible to auto generate signed URLs for image files as they are added to via the storage API using the Postgres HTTP extension, then save them into a public table. I'm already copying the storage table to a public table.

Maybe using the [REST API](https://supabase.github.io/storage-api/#/object/get_object_sign__bucketName___)?

Alternately it seems like we have to call the Supabase-js api on every image file, which could be hundreds in the case of a feed. Are there performance implications for this?

**Comment by silentworks** [comment]
It might be possible to do what you are asking but I haven't tested out the storage API enough as yet to say for sure, but from posts I've seen on here from the Supabase team, they are working on ways to get the signed URLs without you having to make a separate request.

**Reply by cspecter** [reply]
We are using Postgraphile to connect with Supabase and ended up writing a custom resolver to load all the signed URLs. It's not very performant on the first run, but subsequent runs with the same settings it's fairly fast.

**Comment by inian** [comment, marked as answer]
Hey @cspecter, I am working on making entire buckets public [here](https://github.com/supabase/storage-api/pull/19). Should be tested and rolled out in a couple of weeks.

**Reply by cspecter** [reply]
Yes. I think this will help immensely. Our solution is working, but it refreshes the url too often and there is a lot of page flickering.

**Reply by awalias** [reply]
I saw a working demo of this by @inian last night! exciting stuff!

**Reply by inian** [reply]
This has been deployed now.
